debug-tools/snap-debug-info.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+h1(){ echo -e "\n==================== $* ===================="; }
+h2(){ echo -e "\n========== $* =========="; }
+h3(){ echo -e "\n===== $* ====="; }
+
+h1 "SNAP VERSION"; snap version
+h1 "SNAP WHOAMI"; snap whoami
+h1 "SNAP MODEL"; snap model --verbose
+h1 "SNAP MODEL SERIAL"; snap model --serial --verbose
+h1 "SNAP LIST"; snap list
+h1 "SNAP SERVICES"; snap services
+h1 "SNAP CONNECTIONS"; snap connections
+
+h1 "PER-SNAP CONNECTIONS"
+for sn in $(snap list | awk 'NR>1 {print $1}'); do
+    h2 "PER-SNAP $sn CONNECTIONS"
+    snap connections "$sn"
+done
+h1 "SNAP CHANGES"
+snap changes --abs-time
+h1 "GADGET SNAP GADGET.YAML"
+cat /snap/"$(snap list | awk '($6 ~ /.*gadget.*$/) {print $1}')"/current/meta/gadget.yaml
```

Comment:
should we have a `2>&1` in commands that can be expected to fail, in order to keep the stderr clean and log the failure in the output?

Reply:
yes not a bad idea

Reply:
Handled this in a slightly different way